tests/postgres_test.py

```diff
@@ -19,3 +24,37 @@ def test_get_columns_from_table():
         target_schema="public", target_table="test"
     )
     assert columns == expectation
+
+
+@pytest.mark.parametrize(
+    "test_name, schema, table, column_name, result",
+    [
+        (
+            "unique",
+            "schema",
+            "table",
+            "column",
+            [
+                call(
+                    """select count(*) as errors from(
+                select column from schema.table where column is not null group by column having count(*) > 1 )
+                errors"""
+                )
+            ],
+        ),
+        (
+            "not_null",
+            "schema",
+            "table",
+            "column",
+            [call("select count(*) as errors from schema.table where column is null")],
+        ),
+    ],
+)
+def test_run_test(mocker, test_name, schema, table, column_name, result):
+    run_query_test = mocker.patch(
+        "dbt_sugar.core.connectors.postgres_connector.PostgresConnector.run_query_test"
+    )
+    postgres_connector = PostgresConnector(**CREDENTIALS)
+    postgres_connector.run_test(test_name, schema, table, column_name)
+    run_query_test.assert_has_calls(result)
```

Comment:
Actually we don't (and in fact shouldn't) be mocking tests that we do in postgress because we can actually test it fully end to end against the database as it gets set up in docker even on the github actions.

Don't you think it would be a good idea?